047-TrafficControlWithDapr/Student/Challenge-00.md

```diff
@@ -0,0 +1,356 @@
+
+### End-state with Dapr applied
+
+As you complete the challenges, you'll evolve the application architecture to work with Dapr and consume Azure-based backing services:
+
+- Azure IoT Hub
+- Azure Redis Cache
+- Azure Service Bus
+- Azure Logic Apps
+- Azure Key Vault
+
+The following diagram shows the end-state of the application:
+
+<img src="../images/dapr-setup.png" alt="Dapr setup" style="zoom:67%;" />
+
+1. To retrieve driver information using synchronous request/response communication between the `FineCollectionService` and `VehicleRegistrationService`, you'll implement the Dapr **service invocation** building block.
+1. To send speeding violations to the `FineCollectionService`, you'll implement the Dapr **publish and subscribe** building block (asynchronous communication) with the Dapr Azure Service Bus component.
+1. To store vehicle state, you'll implement the Dapr **state management** building block with the Dapr Azure Redis Cache component.
+1. To send fine notices to the owner of a speeding vehicle by email, you'll implement the HTTP **output binding** building block with the Dapr Azure Logic App component.
+1. To send vehicle info to the `TrafficControlService`, you'll use the Dapr **input binding** for MQTT using Dapr Azure IoT Hub component as the MQTT broker.
+1. To retrieve a license key for the fine calculator component and credentials for connecting to the SMTP server, you'll implement the Dapr **secrets management** building block with Dapr Azure Key Vault component.
+
+The following sequence diagram shows how the solution will work after implementing Dapr:
+
+<img src="../images/sequence-dapr.png" alt="Sequence diagram with Dapr" style="zoom:67%;" />
+
+*It's helpful to refer back to the preceding sequence diagram as you progress through the challenges.*
+
+### Prevent port collisions
+
+For most of the challenges, you'll run the microservices in the solution on your local machine. To prevent port collisions, all services will listen on a different HTTP port. When running with Dapr, you need additional ports for HTTP and gRPC communication between the sidecar services. By default, these ports are `3500` and `50001`. However, you'll use different port numbers for each service to prevent collisions. Please closely follow the instructions so that your microservices use the following ports for their Dapr sidecars:
```

Comment:
You mention running the services locally, but rest of instructions only speak to running it in Azure? (by running bicep templates below, etc)  Am I missing anything here?